### PR TITLE
make sentry an optional dependency

### PR DESCRIPTION
**Pull request by robinvd**
## Description

sentry pulls in the whole reqwest (hyper/tokio) depencency tree. With this feature flag those depencies can be avoided.

on my local project (almost empty) this reduces the compile time quite a bit. Using opt-level=1
```
# no-sentry
$ touch src/main.rs && cargo build
Finished dev [unoptimized + debuginfo] target(s) in 16.38s

# sentry
$ touch src/main.rs && cargo build
Finished dev [unoptimized + debuginfo] target(s) in 41.64s
```

## Additions
* Add a feature flag `sentry` to disable the sentry dependency.

## PR Checklist
- [x] Updated the content of the book if this PR would make the book outdated.
- [x] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new code added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [x] Ran `cargo clippy --all`
- [ ] Ran `cargo test --all --features "empty"`

### CODE REVIEW

**Comment by Frizi** [comment]
This looks good as a bandaid, but i'd like to avoid making it appealing to disable sentry logging. We plan to rely on this to detect issues. @fhaynes Do you know if maybe there an alternative way to have sentry but without reqwest?

**Comment by jaynus** [comment]
We can merge this once the CHANGELOG conflict is fixed.

File: Cargo.toml [hunk 1]
```diff
@@ -126,7 +126,7 @@ fern = { version = "0.5", features = ["colored"] }
 log = { version = "0.4.6", features = ["serde"] }
 rayon = "1.0.2"
 rustc_version_runtime = "0.1"
-sentry = "0.15.4"
```
Comment (ishitatsuyuki) [inline Cargo.toml]: Nit: whitespace missing

**Comment by jaynus** [comment]
@robinvd Can you fix the changelog conflict with master so we can get this merged?

**Comment by azriel91** [comment]
Rebased as #1825.

**Comment by robinvd** [comment]
@jaynus sorry for missing the ping.

**Pull request closed without merging**